components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/EndpointAdminServiceProxy.java

```diff
@@ -154,14 +154,12 @@ public boolean isEndpointExist(String endpointName)
             throws EndpointAdminException {
 
         if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
-            String[] endPointsNames = endpointAdmin.getEndPointsNames();
-            if (endPointsNames != null) {
-                return Arrays.asList(endPointsNames).contains(endpointName);
+            if (endpointAdmin.isEndpointExist(endpointName)) {
+                return true;
```

Comment:
can do in one line

Reply:
https://github.com/wso2/carbon-apimgt/pull/9174/commits/456ff9aca0a18f37ab9fb6a9f1bd92957463f016